poulpe-model/src/test/java/org/jtalks/poulpe/model/dao/hibernate/UserHibernateDaoTest.java

```diff
@@ -97,6 +104,69 @@ public void testGetByUsername() {
 	}
 
     @Test
+    public void shouldReturnListWithSingleUserForUniqueUsername() throws Exception {
+        PoulpeUser poulpeUser = saveAndEvictUser(USERNAME, PASSWORD);
+        saveAndEvictUser(english(5,20), PASSWORD);
+
+        List<PoulpeUser> poulpeUsers = dao.getByUsernameAndPasswordHash(USERNAME, PASSWORD);
+
+        assertEquals(poulpeUsers.size(), 1);
+        assertEquals(poulpeUsers.get(0), poulpeUser);
+    }
+
+    @Test
+    public void shouldReturnListWithAllUsersMatchedByUsername() throws Exception {
+        List<PoulpeUser> expectedUsersList = Arrays.asList(
+                saveAndEvictUser(USERNAME, PASSWORD),
+                saveAndEvictUser(USERNAME.toLowerCase(), PASSWORD),
+                saveAndEvictUser(StringUtils.capitalize(USERNAME), PASSWORD),
+                saveAndEvictUser(USERNAME.toUpperCase(), PASSWORD)
+        );
+        PoulpeUser anotherUser = saveAndEvictUser(english(5, 20), PASSWORD);
+
+        List<PoulpeUser> poulpeUsers = dao.getByUsernameAndPasswordHash(USERNAME, PASSWORD);
+
+        assertEquals(poulpeUsers.size(), 4);
+        assertTrue(poulpeUsers.containsAll(expectedUsersList));
+        assertFalse(poulpeUsers.contains(anotherUser));
+    }
+
+    @Test
+    public void shouldReturnEmptyListIfUsernameOrPasswordIsEmpty() throws Exception {
+        saveAndEvictUser(USERNAME, PASSWORD);
+
+        List<PoulpeUser> poulpeUsers = dao.getByUsernameAndPasswordHash("", PASSWORD);
+        assertTrue(poulpeUsers.isEmpty());
+
+        poulpeUsers = dao.getByUsernameAndPasswordHash(USERNAME, "");
+        assertTrue(poulpeUsers.isEmpty());
+    }
+
+    @Test
+    public void shouldReturnEmptyListIfUserNotFound() throws Exception {
+        saveAndEvictUser(USERNAME, PASSWORD);
+
+        List<PoulpeUser> poulpeUsers = dao.getByUsernameAndPasswordHash(alphanumeric(10), alphanumeric(10));
+
+        assertTrue(poulpeUsers.isEmpty());
```

Comment:
While this looks nice, it's more practical to have `assertEquals(0, poulpeUsers.size())` - you'd see how many were actually there in this case.